cmake/Modules/FindMKLDNN.cmake

```
@@ -70,7 +70,11 @@ IF(MKLDNN_USE_NATIVE_ARCH)  # Disable HostOpts in MKLDNN unless MKLDNN_USE_NATIV
   SET(MKLDNN_ARCH_OPT_FLAGS "HostOpts" CACHE STRING "" FORCE)
 ELSE()
   IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    SET(MKLDNN_ARCH_OPT_FLAGS "-msse4" CACHE STRING "" FORCE)
+    IF(CPU_AARCH64)
+      SET(MKLDNN_ARCH_OPT_FLAGS "-mcpu=native" CACHE STRING "" FORCE)
+    ELSE()
+      SET(MKLDNN_ARCH_OPT_FLAGS "-msse4" CACHE STRING "" FORCE)
+    ENDIF()
```

Nit (`-mcpu=native` is default setting, isn't it, so there are no need to set it explicitly)
```suggestion
    IF(CPU_INTEL)
      SET(MKLDNN_ARCH_OPT_FLAGS "-msse4" CACHE STRING "" FORCE)
    ENDIF()
```

---

Actually I believe the default is equivalent to `-mcpu=generic` to generate portable code - recommendation is to use `-mcpu=native` for optimised code - https://community.arm.com/developer/tools-software/tools/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu

However, this doesn't entirely makes sense for cross compiling - perhaps this should only be set in the case that `CMAKE_HOST_SYSTEM_PROCESSOR` matches `CMAKE_SYSTEM_PROCESSOR`, as would be the case for a 'stand alone` oneDNN build: https://github.com/oneapi-src/oneDNN/blob/2181e7e21a9cd68848d775e53bcf8994941adb61/cmake/platform.cmake#L126

---

True, but this is a desired default behavior, as we want to build an optimized by portable binaries by default (this is why on Intel platform `-msee4` flag is used instead of `-mavx512f`, as default binaries should be compatible with a decade old Xeon CPUs that are still common in some server farms.

User can always override default C++ flags when compiling on Linux by passing `CXXFLAGS` environment variable, which could contain `-mcpu=native`

---

Yes, that's a fair point.